### ISSUE TITLE
Bug: Deadlock caused by AzVolume finalizer 

### ISSUE THREAD

**Issue by landreasyan** [abhisheksinghbaghel/azuredisk-csi-driver#164]
**What happened**:
When deleting resources with persistent volumes, the AzVolume controller sometimes fails to delete the finalizers on AzVolume CRIs. This makes the resource impossible to delete without a manual patch and thus deadlocks the deletion of the namespace which can later cause a failure to uninstall/delete the driver. 

**What you expected to happen**:
All finalizers on AzVolume and AzVolumAttachments should be removed programmatically when persistent volumes are deleted. There is a bug in azure cloud provider that causes deletion of persistent volume to fail due to the volume still being attached to a node. This happens even though the detach request has previously reported a success. We either need to fix the correct status reporting for detach requests in azure cloud provider or find a workaround to remove the finalizers from AzVolumes with this failure to avoid deadlock during driver uninstall or namespace deletion.

**How to reproduce it**:
I have seen it most often happen when I run the [failover scheduler extender test ](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/2a48fce638545faac470663716b575a194edfacf/test/e2e/azdiskschedulerextender_test.go#L204)that creates and deletes multiple pod with many persistent volumes. 

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by sunpa93** [comment]
Currently, there is a bug related with the `cloud-provider-azure` vendor pkg that returns success on detach operation that actually failed. This problem has been resolved in their repo's more recent commits and has been updated in our upstream's repo as well.

Say `controllerCommon.DetachDiskfor` disk A, B, C on same node are made. A acquires the lock first, and `vmSet.DetachDisk` is called for A. While the detach operation for the A is being executed, the other two disks will be put into detachment request map. Then, when the vmss update completes and lock is lifted, either B or C will acquire the lock. Say B did. This call will call `vmSet.DetachDisk` on both B and C and empty the detachment request map. And if this vmss update fails for whatever reason, `controllerCommon.DetachDisk` for B will rightfully return an error but `controllerCommon.DetachDisk` for C will return success when it should have returned an error because the detachment request map would be empty from `controllerCommon.DetachDisk` for B.


Locally testing with the changes merged, I was able to confirm that the abovementioned problem had been resolved.

That said, we can expect that this problem to be resolved along with the coming iteration of upstream merge.
